roles/rsyslog/tasks/main.yml

```diff
@@ -130,37 +130,80 @@
       when: (rsyslog_enabled | bool)
       notify: restart rsyslogd
 
-    - name: Run deploy input sub-roles configs
-      include_role:
-        name: "{{ role_path }}/roles/input_roles/{{ input_item.type }}"
+    - name: Include input sub-vars 
+      include_vars:
+        file: "{{ varfile }}"
+      vars:
+        varfile: "inputs/{{ input_item.type }}.yml"
+        varfile_abspath: "{{ role_path }}/vars/{{ varfile }}"
       loop: "{{ logging_inputs | flatten(1) }}"
       loop_control:
         loop_var: input_item
       when:
         - rsyslog_enabled | bool
         - input_item | d([])
-        - input_item.state | d('present') != 'absent'
+        - varfile_abspath is exists
+      tags:
+        - skip_ansible_lint
+
+    - name: Run input sub-tasks 
+      include_tasks:
+        file: "{{ tasks }}"
+      vars:
+        tasks: "inputs/{{ input_item.type }}.yml"
+        tasks_abspath: "{{ role_path }}/tasks/{{ tasks }}"
+      loop: "{{ logging_inputs | flatten(1) }}"
+      loop_control:
+        loop_var: input_item
+      when:
+        - rsyslog_enabled | bool
+        - input_item | d([])
+        - tasks_abspath is exists
+      tags:
+        - skip_ansible_lint
 
-    - name: Run remove input sub-roles configs
-      include_role:
-        name: "{{ role_path }}/roles/input_roles/{{ input_item.type }}"
-        tasks_from: cleanup.yml
+    - name: Run remove input sub-tasks
+      include_tasks:
+        file: "{{ tasks }}"
+      vars:
+        tasks: "inputs/{{ input_item.type }}_cleanup.yml"
```

Comment:
I would create a subdirectory for every former input subrole, so make `inputs/{{ input_item.type }}_cleanup.yml` into `inputs/{{ input_item.type }}/cleanup.yml` (and `inputs/{{ input_item.type }}.yml` into `inputs/{{ input_item.type }}/main.yml`)